.github/workflows/ci-tests.yml

```diff
@@ -87,7 +87,7 @@ jobs:
 
   build-and-unit-test:
     runs-on: macos-13
-    needs: [tuist-generation, changes]
+    needs: [tuist-generation, changes, verify-frontend-bundle-latest]
```

Comment:
Will this optimization end up causing CI run times to be longer more often than it's worth? Or is that job quick enough that it shouldn't matter?

Reply:
Here's a list of all the PRs that have had that job run since it was added; averaged across all of them it comes to 31s.
* https://github.com/apollographql/apollo-ios-dev/actions/runs/7190250906 (27s)
* https://github.com/apollographql/apollo-ios-dev/actions/runs/7190133181 (37s)
* https://github.com/apollographql/apollo-ios-dev/actions/runs/7145358499 (24s)
* https://github.com/apollographql/apollo-ios-dev/actions/runs/7145269481 (26s)
* https://github.com/apollographql/apollo-ios-dev/actions/runs/7124028071 (47s)
* https://github.com/apollographql/apollo-ios-dev/actions/runs/7124001038 (27s)

9 out of 10 times there will probably be no change to the bundle so if you add 31s to each of the many many PRs we'd have run before an unintentional bundle change occurs it's probably more than the single PR that could have exited early. So maybe we don't actually need to try optimize for this.

Reply:
I decided to revert it. I expect the PR that gets caught in the bundle validation job will be far less frequent than all the other PRs that needlessly wait for it to pass.